### DISCUSSION TITLE
Login to Tidal using Tidal SDK Android doesn't work from an emulator

### DISCUSSION THREAD

**Discussion by ForMiSoft** [tidal-music/discussions#108] [Q&A] [answered]
My Android app is now able to login, but only from a physical phone, the emulator in Android Studio still says it's not logged in. Is that by a design or is it an error?

**Comment by aliaksei-taliuk** [comment]
@ForMiSoft I forwarded your question to the corresponding team. My colleagues will try to assist you as soon as they have capacity for that. Thanks!

**Comment by michpohl** [comment, marked as answer]
The SDK auth module does not change behavior based on if you use an emulator or a real device. So it's probably an implementation detail in your app. 
If you think the behavior is caused by a bug on our end, please provide sufficient information for us to understand the problem, such as logs, code examples, etc.
Please also note that if you find further problems with the SDK, you should create an issue directly in [the repo's issues](https://github.com/tidal-music/tidal-sdk-android/issues), to reach the right people faster.

**Reply by ForMiSoft** [reply]
Thanks.   The same app can login successfully on my phone (Samsung Galaxy S10) but when I run it in the emulator (Pixel 6) it ends up as not logged in.  I have no idea how this is possible and I don't know where to look for the cause, so I guess I'll leave it at that. If you are interested in looking at it, I can add you to my repository again. If not, you can close this thread as solved.

**Reply by michpohl** [reply]
Please investigate on your end, and - if it turns out to be unfixable on your end - report back with data illustrating your case. It is unfortunately not a sustainable practice to help debugging by downloading and running user repos. Thank you for your understanding!

**Reply by ForMiSoft** [reply]
That's probably a misunderstanding.  In this case, I don't need any help with my application.  The app works and that's enough for me.  I offered you the opportunity to reveal why it doesn't work on the emulator, I myself have no interest and no reason to look for why the login doesn't work on the emulator.  I'm also not happy to give you access to my repositories, but I wanted to help you.  If you're not interested, that's ok and I'll consider it closed.
